extension/content/components/recipes/form/filters/FallbackFO.js

```diff
@@ -0,0 +1,59 @@
+// @ts-nocheck
+import React from "react";
+import { Badge, ControlLabel, FormGroup } from "rsuite";
+
+import JsonEditor from "devtools/components/common/JsonEditor";
+import {
+  ACTION_UPDATE_DATA,
+  useRecipeDetailsData,
+  useRecipeDetailsDispatch,
+} from "devtools/contexts/recipeDetails";
+import { partitionFO } from "devtools/components/recipes/form/filters/partitionFO";
+
+export default function FallbackFO() {
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+  const [invalidJSON, setInvalidJson] = React.useState(false);
+
+  let knownFO = [];
+  let additionalFO = [];
+  if (data.filter_object) {
+    [knownFO, additionalFO] = partitionFO(data.filter_object);
+  }
+
+  const handleChange = (value) => {
+    if (value) {
+      [knownFO, additionalFO] = partitionFO(data.filter_object);
+      const newFilterObjects = [...knownFO, ...value];
+
+      dispatch({
+        type: ACTION_UPDATE_DATA,
+        data: {
+          ...data,
+          filter_object: newFilterObjects,
+        },
+      });
+      setInvalidJson(false);
+    } else {
+      setInvalidJson(true);
+    }
+  };
+
+  const inValidBadge = () => {
```

Comment:
nit: `inValid` is strange camelcasing when the word is `invalid` and not `in valid`.

Reply:
also i dont think this needs to be a function? it could simply be written as:
```js
let invalidBadge = null;
if (invalidJSON) {
  invalidBadge = <Badge content="Invalid JSON" />;
}
```